docs/guides/service_configuration_file.md

```diff
@@ -152,143 +168,197 @@ type: skill
-    benchmark_tool:
-      args: *id001
-```
+    If you wish that each agent outputs a different message in the [Hello World service](../demos/hello_world_demo.md) with four agents, you can define the following multiple override in the `service.yaml` file:
 
+    ```yaml
+    (...)
+    ---
+    public_id: valory/hello_world_abci:0.1.0
+    type: skill
+    0:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 0
+    1:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 1
+    2:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 2
+    3:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 3
+    ```
 
-### Overrides with complicated data structures
+If you have repetitive overridden parameters, you can define them using [YAML anchors](https://yaml.org/spec/1.2.2/) under the section `extra`, and reference them when defining overrides for different agents.
 
-If you have nested list the environment export rules will differ as per the inner data structures. If the list is a strict list, meaning the inner values are either primitive data types like `int`, `str`, `float` or `boolean`, the list will be exported as a `json` string.
+???+ example
 
+    In this example, we define a YAML anchor with label `&id001` to avoid repeating the same configuration of the `args` parameter in all the agents.
 
-```yaml
-(...)
----
-extra:
-  benchmark_tool:
-    args: &id001
-      log_dir: /benchmarks
-public_id: valory/hello_world_abci:0.1.0
-type: skill
-models:
-  params:
-    args:
-      simple_list:
-        - 1
-        - 2
-        - 3
-      nested_list:
-      - - foo
-        - bar
-      - - "hello"
-        - "world"
-```
+    ```yaml
+    (...)
+    ---
+    extra:
+      benchmark_tool:
+        args: &id001
+          log_dir: /benchmarks
+    public_id: valory/hello_world_abci:0.1.0
+    type: skill
+    0:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 0
+        benchmark_tool:
+          args: *id001
+    1:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 1
+        benchmark_tool:
+          args: *id001
+    2:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 2
+        benchmark_tool:
+          args: *id001
+    3:
+      models:
+        params:
+          args:
+            hello_world_message: Hello, from agent 3
+        benchmark_tool:
+          args: *id001
+    ```
 
-These parameters will be exported as
+## Export to environment variables
 
+The framework exports each overridden attribute in the service configuration file to a system environment variable. By default, the exported environment variables are labelled with their upper case JSON path:
```

Comment:
by default? What is the alternative way to export them?

Reply:
Fixed